compute/client_library/snippets/tests/test_compute_reservation.py

```diff
@@ -128,3 +124,27 @@ def test_list_compute_reservation(reservation):
 def test_delete_compute_reservation(reservation):
     response = delete_compute_reservation(PROJECT_ID, ZONE, reservation.name)
     assert response.status == Operation.Status.DONE
+
+
+def test_create_shared_reservation():
+    """Test for creating a shared reservation.
+
+    The reservation will be created in PROJECT_ID and shared with the project specified
+    by SHARED_PROJECT_ID.
+
+    Make sure to set the SHARED_PROJECT_ID environment variable before running this test,
+    and ensure that the project is allowlisted in the organization policy for shared reservations.
+
+    If the SHARED_PROJECT_ID environment variable is not set, the test will be skipped.
+    """
+    if not SHARED_PROJECT_ID:
+        pytest.skip(
+            "Skipping test because SHARED_PROJECT_ID environment variable is not set."
+        )
+    try:
+        response = create_compute_shared_reservation(
+            PROJECT_ID, ZONE, RESERVATION_NAME, SHARED_PROJECT_ID
+        )
+        assert response.share_settings.project_map.values()
+    finally:
+        delete_compute_reservation(PROJECT_ID, ZONE, RESERVATION_NAME)
```

Comment:
use another try and catch to not report an error when the reservation is missing (if creation failed)

Reply:
Added🙌